### ISSUE TITLE
docs: don't do `mul!(x, A, x)`

### ISSUE THREAD

**Issue by vpuri3** [SciML/LinearSolve.jl#97]
https://github.com/SciML/LinearSolve.jl/blob/43111fc88f7d0f1ab1b32460c2f9176341e1959f/src/preconditioners.jl#L29

docs:
```julia
Y <- mul!(Y, A, B)
```
Calculates the matrix-matrix or matrix-vector product AB and stores the result in Y, overwriting the existing value of Y. Note that Y must not be aliased with either A or B.

**Comment by ChrisRackauckas** [comment]
See if tests pass when that one is deleted.

**Comment by vpuri3** [comment]
yeah applying the inverse operation of an inverted operator is behaviour that should be avoided.
